src/test/java/com/kodilla/ecommercee/domain/CartTestSuite.java

```diff
@@ -0,0 +1,190 @@
+        long cartId = cart.getCartId();
+
+        //Then
+        BigDecimal value = new BigDecimal("0");
+
+        assertNotEquals(0, cartId);
+        assertEquals(0, products.size());
+        assertEquals(value, cart.getValue());
+    }
+
+    @Test
+    public void shouldAddOneProduct() {
+        //Given
+        Group group = new Group("Toys");
+        Product product = new Product("Teddy Bear", "little plush toy", new BigDecimal("10"));
+        group.getProducts().add(product);
+        product.getGroupsOfProduct().add(group);
+
+        Cart cart = new Cart(new BigDecimal("0"));
+        cart.getProducts().add(product);
+        cart.setValue(new BigDecimal("10"));
+        List<Product> products = cart.getProducts();
+
+        //When
+        cartRepository.save(cart);
+        long cartId = cart.getCartId();
+
+        //Then
+        assertNotEquals(0, cartId);
+        assertEquals(1, products.size());
+        assertEquals(product, cart.getProducts().get(0));
+    }
+
+    @Test
+    public void shouldAddSeveralOfTheSameProductsToCart(){
+        //Given
+        Group group = new Group("Toys");
+        Product product = new Product("blocks", "wooden blocks", new BigDecimal("4.50"));
+        group.getProducts().add(product);
+        product.getGroupsOfProduct().add(group);
+
+        Cart cart = new Cart();
+        cart.getProducts().add(product);
+        cart.getProducts().add(product);
+        cart.getProducts().add(product);
+        cart.getProducts().add(product);
+        cart.getProducts().add(product);
+        cart.setValue(new BigDecimal("22.50"));
+        List<Product> products = cart.getProducts();
+
+        //When
+        cartRepository.save(cart);
+        long cartId = cart.getCartId();
+
+        //Then
+        assertNotEquals(0, cartId);
+        assertEquals(5, products.size());
+        assertEquals(cart.getProducts().get(0), cart.getProducts().get(3));
+        assertEquals("blocks", cart.getProducts().get(0).getProductName());
+    }
+
+    @Test
+    public void  shouldDifferentiateBetweenProductsWithSameNameDifferentPrice() {
+        //Given
+        Group group = new Group("Toys");
+        Product product1 = new Product("blocks", "wooden blocks", new BigDecimal("4.50"));
+        Product product2 = new Product("blocks", "wooden blocks", new BigDecimal("20"));
+        group.getProducts().add(product1);
+        group.getProducts().add(product2);
+        product1.getGroupsOfProduct().add(group);
+        product2.getGroupsOfProduct().add(group);
+
+        Cart cart = new Cart();
+        cart.getProducts().add(product1);
+        cart.getProducts().add(product2);
+        cart.setValue(new BigDecimal("24.50"));
+        List<Product> products = cart.getProducts();
+
+        //When
+        cartRepository.save(cart);
+        long cartId = cart.getCartId();
+
+        //Then
+        assertNotEquals(0, cartId);
+        assertNotEquals(product1, product2);
```

Comment:
To sprawdzenie jest nie potrzebne, bo tutaj sprawdzasz dwa obiekty utworzne wcześniej